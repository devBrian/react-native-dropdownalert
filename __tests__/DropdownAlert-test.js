import 'react-native';
import React from 'react';
import DropdownAlert from '../DropdownAlert';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

test('renders info alert correctly', () => {
  const wrapper = shallow(<DropdownAlert imageSrc={'https://facebook.github.io/react/img/logo_og.png'} />);
  wrapper.setState({isOpen: true, type: 'info', title: 'Hello', message: 'World!'});
  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});
test('renders warn alert correctly', () => {
  const wrapper = shallow(<DropdownAlert imageSrc={'https://facebook.github.io/react/img/logo_og.png'} />);
  wrapper.setState({isOpen: true, type: 'warn', title: 'Hello', message: 'World!'});
  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});
test('renders error alert correctly', () => {
  const wrapper = shallow(<DropdownAlert imageSrc={'https://facebook.github.io/react/img/logo_og.png'} />);
  wrapper.setState({isOpen: true, type: 'error', title: 'Hello', message: 'World!'});
  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});
test('renders success alert correctly', () => {
  const wrapper = shallow(<DropdownAlert imageSrc={'https://facebook.github.io/react/img/logo_og.png'} />);
  wrapper.setState({isOpen: true, type: 'success', title: 'Hello', message: 'World!'});
  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});
test('renders custom alert correctly', () => {
  const wrapper = shallow(<DropdownAlert imageSrc={'https://facebook.github.io/react/img/logo_og.png'} />);
  wrapper.setState({isOpen: true, type: 'custom', title: 'Hello', message: 'World!'});
  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});
test('renders custom alert with cancel correctly', () => {
  const wrapper = shallow(<DropdownAlert imageSrc={'https://facebook.github.io/react/img/logo_og.png'} showCancel={true} />);
  wrapper.setState({isOpen: true, type: 'custom', title: 'Hello', message: 'World!'});
  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});
