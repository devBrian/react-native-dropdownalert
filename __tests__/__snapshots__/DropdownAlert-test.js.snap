// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders custom alert correctly 1`] = `
<AnimatedComponent
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "transform": Array [
        Object {
          "translateY": -100,
        },
      ],
    }
  }
>
  <TouchableHighlight
    activeOpacity={0.85}
    disabled={false}
    onLayout={[Function]}
    onPress={[Function]}
    underlayColor="black"
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "padding": 8,
            "paddingTop": 20,
          },
          Object {
            "flexDirection": "row",
            "padding": 16,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": "https://facebook.github.io/react/img/logo_og.png",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "height": 36,
            "padding": 8,
            "width": 36,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "padding": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlign": "left",
            }
          }
        >
          Hello
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={3}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "white",
              "fontSize": 14,
              "fontWeight": "normal",
              "textAlign": "left",
            }
          }
        >
          World!
        </Text>
      </View>
    </View>
  </TouchableHighlight>
</AnimatedComponent>
`;

exports[`renders custom alert with cancel correctly 1`] = `
<AnimatedComponent
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "transform": Array [
        Object {
          "translateY": -100,
        },
      ],
    }
  }
>
  <TouchableHighlight
    activeOpacity={0.85}
    disabled={false}
    onLayout={[Function]}
    onPress={null}
    underlayColor="black"
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "padding": 8,
            "paddingTop": 20,
          },
          Object {
            "flexDirection": "row",
            "padding": 16,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": "https://facebook.github.io/react/img/logo_og.png",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "height": 36,
            "padding": 8,
            "width": 36,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "padding": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlign": "left",
            }
          }
        >
          Hello
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={3}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "white",
              "fontSize": 14,
              "fontWeight": "normal",
              "textAlign": "left",
            }
          }
        >
          World!
        </Text>
      </View>
      <TouchableHighlight
        activeOpacity={0.85}
        onPress={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "height": 36,
            "width": 36,
          }
        }
        underlayColor="black"
      >
        <Image
          source={1}
          style={
            Object {
              "alignSelf": "center",
              "height": 36,
              "padding": 8,
              "width": 36,
            }
          }
        />
      </TouchableHighlight>
    </View>
  </TouchableHighlight>
</AnimatedComponent>
`;

exports[`renders error alert correctly 1`] = `
<AnimatedComponent
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "transform": Array [
        Object {
          "translateY": -100,
        },
      ],
    }
  }
>
  <TouchableHighlight
    activeOpacity={0.85}
    disabled={false}
    onLayout={[Function]}
    onPress={[Function]}
    underlayColor="#cc3232"
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "padding": 8,
            "paddingTop": 20,
          },
          Object {
            "backgroundColor": "#cc3232",
          },
        ]
      }
    >
      <Image
        source={1}
        style={
          Object {
            "alignSelf": "center",
            "height": 36,
            "padding": 8,
            "width": 36,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "padding": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlign": "left",
            }
          }
        >
          Hello
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={3}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "white",
              "fontSize": 14,
              "fontWeight": "normal",
              "textAlign": "left",
            }
          }
        >
          World!
        </Text>
      </View>
    </View>
  </TouchableHighlight>
</AnimatedComponent>
`;

exports[`renders info alert correctly 1`] = `
<AnimatedComponent
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "transform": Array [
        Object {
          "translateY": -100,
        },
      ],
    }
  }
>
  <TouchableHighlight
    activeOpacity={0.85}
    disabled={false}
    onLayout={[Function]}
    onPress={[Function]}
    underlayColor="#2B73B6"
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "padding": 8,
            "paddingTop": 20,
          },
          Object {
            "backgroundColor": "#2B73B6",
          },
        ]
      }
    >
      <Image
        source={1}
        style={
          Object {
            "alignSelf": "center",
            "height": 36,
            "padding": 8,
            "width": 36,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "padding": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlign": "left",
            }
          }
        >
          Hello
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={3}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "white",
              "fontSize": 14,
              "fontWeight": "normal",
              "textAlign": "left",
            }
          }
        >
          World!
        </Text>
      </View>
    </View>
  </TouchableHighlight>
</AnimatedComponent>
`;

exports[`renders success alert correctly 1`] = `
<AnimatedComponent
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "transform": Array [
        Object {
          "translateY": -100,
        },
      ],
    }
  }
>
  <TouchableHighlight
    activeOpacity={0.85}
    disabled={false}
    onLayout={[Function]}
    onPress={[Function]}
    underlayColor="#32A54A"
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "padding": 8,
            "paddingTop": 20,
          },
          Object {
            "backgroundColor": "#32A54A",
          },
        ]
      }
    >
      <Image
        source={1}
        style={
          Object {
            "alignSelf": "center",
            "height": 36,
            "padding": 8,
            "width": 36,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "padding": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlign": "left",
            }
          }
        >
          Hello
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={3}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "white",
              "fontSize": 14,
              "fontWeight": "normal",
              "textAlign": "left",
            }
          }
        >
          World!
        </Text>
      </View>
    </View>
  </TouchableHighlight>
</AnimatedComponent>
`;

exports[`renders warn alert correctly 1`] = `
<AnimatedComponent
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "transform": Array [
        Object {
          "translateY": -100,
        },
      ],
    }
  }
>
  <TouchableHighlight
    activeOpacity={0.85}
    disabled={false}
    onLayout={[Function]}
    onPress={[Function]}
    underlayColor="#cd853f"
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "padding": 8,
            "paddingTop": 20,
          },
          Object {
            "backgroundColor": "#cd853f",
          },
        ]
      }
    >
      <Image
        source={1}
        style={
          Object {
            "alignSelf": "center",
            "height": 36,
            "padding": 8,
            "width": 36,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "padding": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlign": "left",
            }
          }
        >
          Hello
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={3}
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "white",
              "fontSize": 14,
              "fontWeight": "normal",
              "textAlign": "left",
            }
          }
        >
          World!
        </Text>
      </View>
    </View>
  </TouchableHighlight>
</AnimatedComponent>
`;
